heartbeat/monitors/monitor.go

```diff
@@ -105,6 +119,32 @@ func newMonitor(
 	}
 
 	jobs, endpoints, err := monitorPlugin.create(config)
+
+	// Set the correct monitor.id for all jobs
+	var idWrapper func(job Job) Job
+	if m.id != "" {
+		// Ensure we don't have duplicate IDs
```

Comment:
If the users tries to have the same config twice, he gets an error? In MB / FB if we have the exact same config twice I think we just ignore it (not 100% sure).

Functionality wise I was thinking internally we always use the `hash` as the job id for the registry and only enrich it / overwrite it with the user generated id when the event is sent.

Reply:
I think if a user is doing this it is a bug they'd want to know about. There's no legitimate reason to do this no? Given the convo we had about the ID being unique to a given config on a hb instance does this comment still stand?

Reply:
I think you are right that there is no reason a user wants to have it twice. We should probably also fail in the other Beats if this happens. 

One reason to skip the error but still report is to make sure all other configs still get started even if one is wrong.

Reply:
OK, this is fixed in https://github.com/elastic/beats/pull/9697/commits/b15829790feaad1803ad5ba139c736262eb07c0e

So, the error is still here, but if you use this in a dynamically loaded monitor it is not fatal.

It *is* still fatal if you define the conflicting monitor in `heartbeat.yml` but I think that's correct, since that file cannot be fixed without a restart.